src/projections.jl

```diff
@@ -152,6 +153,92 @@ function vec_symm(X)
     return X[LinearAlgebra.triu(trues(size(X)))]
 end
 
+"""
+    projection_on_set(::DefaultDistance, v::AbstractVector{T}, ::MOI.ExponentialCone) where {T}
+
+projection of vector `v` on closure of the exponential cone
+i.e. `cl(Kexp) = {(x,y,z) | y e^(x/y) <= z, y>0 } U {(x,y,z)| x <= 0, y = 0, z >= 0}`.
+
+References:
+* [Proximal Algorithms, 6.3.4](https://web.stanford.edu/~boyd/papers/pdf/prox_algs.pdf)
+by Neal Parikh and Stephen Boyd.
+* [Projection, presolve in MOSEK: exponential, and power cones]
+(https://docs.mosek.com/slides/2018/ismp2018/ismp-friberg.pdf) by Henrik Friberg
+"""
+function projection_on_set(::DefaultDistance, v::AbstractVector{T}, s::MOI.ExponentialCone) where {T}
+    _check_dimension(v, s)
+
+    if distance_to_set(DefaultDistance(), v, MOI.ExponentialCone()) < 1e-8
```

Comment:
I think distances should use projections more than the other way around.

Reply:
What's the advantage of using projections for distances?

If we fall into case 4 for the exponential cone projection, i.e., (x,y,z) not in K, -K*, and (x > 0 or y > 0), then computing the projection is going to take significantly longer than computing the distance. A few microseconds vs 10s of nanoseconds.

Reply:
> What's the advantage of using projections for distances?

distance can be seen as a composition of the projection with a point-to-point distance. What we need here to be precise is the set membership for the exp cone and its polar. I advised using distance for this and checking distance approx 0.

@joaquimg do you think we should use another membership function here? The motivation for re-using distance was avoiding duplication

Reply:
This might be the first version of having two distances.
Because a very common distance is the euclidean.
But euclidean requires the projection first.
Our current distance is some epigraphical distance which is good for boolean checking if the point is the the cone.
I don't we should remove the current distance, I would just rename them and have both.
One could be used inside the the other, but I would use euclidean by default. Actually, whenever I possible I think we should use euclidean to have is as uniform as possible.

Reply:
@tjdiamandis to simplify your work for now, you can just implement the quick check _in_exponential(v) that you had before.
Sorry for making you backtrack on this. I'll refactor it with the additional distance in another PR

Reply:
No worries! Sounds good. Will do today.

Reply:
We can just open an issue and let it be like this for now.

Reply:
see here: https://github.com/matbesancon/MathOptSetDistances.jl/issues/32